### PR TITLE
#166153570: Event host should be able to connect the App to slack explicitly

### PR DESCRIPTION
**Pull request by njeri-ngigi**
#### What Does This PR Do?
- adds a modal that has instructions to connect to slack

#### Description Of Task To Be Completed
- As an event host, I should be able to explicitly connect the Andela Social app to Slack. After creating an event, if I am not already connected, then I should see a modal that prompts me to connect the Andela social app to Slack.

#### Any Background Context You Want To Provide?
- N/A

#### How can this be manually tested?
- set SLACK_AUTH_REDIRECT_URI in your environment variable to `${SECURE_FRONTEND_BASE_URL}/slacktokencallback`. P:S: This URL must be included as one of the Redirect URLs in the slack settings.
- on the frontend, create a new event.
- if a slack token isn't found on the db the response will trigger the slack modal to pop up
- if you click cancel, the modal disappears
- if you click on confirm, you are redirected to a slack auth page to confirm
- you will be redirected back to the frontend and a response toast message will appear

#### What are the relevant pivotal tracker stories?
[#166153570 Event host should be able to connect the App to slack explicitly](https://www.pivotaltracker.com/story/show/166153570)

#### Screenshot(s)
image

image

### CODE REVIEW

**Comment by brianbirir** [comment]
@njeri-ngigi thanks for this. Kindly resolve the merge conflict first.

File: server/graphql_schemas/event/schema.py [hunk 1]
```diff
@@ -113,8 +114,14 @@ def mutate_and_get_payload(cls, root, info, **input):
         except ValueError as e:
             raise GraphQLError("An Error occurred. \n{}".format(e))
 
+        slack_token = False
+        if (user_profile.slack_token):
+            slack_token = Truegst
```
Comment (Alameen688) [inline server/graphql_schemas/event/schema.py]: Nice work @njeri-ngigi. It seems there is a typo here. Kindly help fix that. Thanks
```suggestion
            slack_token = True
```

Comment (njeri-ngigi) [reply server/graphql_schemas/event/schema.py]: thanks for catching that

**Review by Alameen688 — COMMENTED** [review]
Great work on the implementation @nj3ry. However, the slack token doesn’t get saved to the database yet. It seems you still need to run a query (with the code sent from slack) against the `CreateUserAuth` [mutation](https://github.com/AndelaOSP/Andela-Socials/blob/e0bff21351cdb1456208017709dc8d2a24b215e2/server/graphql_schemas/users/schema.py#L23) which exchanges the code for the slack_token and saves it. Kindly help look into that.

If there is anything I'm missing, please help clarify. Thank you.

If need be, you can also sync with @Edward-K1 as he implemented the backend of this task at [#184](https://github.com/AndelaOSP/Andela-Socials/pull/184)

**Review by brianbirir — COMMENTED** [review]
Looks good. Ready for merge

**Pull request merged**